contracts/PercentageTransferManager/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+        pub max_allowed_percentage: storage::Value<u128>,
+        /// By toggling the primary issuance variable it will bypass
+        /// all the restrictions imposed by this smart extension
+        pub allow_primary_issuance: storage::Value<bool>,
+        /// Exemption list that contains the list of investor's identities
+        /// which are not affected by this module restrictions
+        pub exemption_list: storage::HashMap<IdentityId, bool>,
+    }
+
+    #[ink(event)]
+    struct ChangeAllowedPercentage {
+        #[ink(topic)]
+        old_percentage: u128,
+        #[ink(topic)]
+        new_percentage: u128,
+    }
+
+    #[ink(event)]
+    struct ChangePrimaryIssuance {
+        #[ink(topic)]
+        allow_primary_issuance: bool,
+    }
+
+    #[ink(event)]
+    struct ModifyExemptionList {
+        #[ink(topic)]
+        identity: IdentityId,
+        #[ink(topic)]
+        exempted: bool,
+    }
+
+    #[ink(event)]
+    struct TransferOwnership {
+        #[ink(topic)]
+        new_owner: AccountId,
+        #[ink(topic)]
+        old_owner: AccountId,
+    }
+
+    impl PercentageTransferManagerStorage {
+        /// Constructor that initializes the `u128` value to the given `max_allowed_percentage`,
+        /// boolean value for the `allow_primary_issuance` & `owner` of the SE.
+        #[ink(constructor)]
+        fn new(&mut self, max_percentage: u128, primary_issuance: bool) {
+            self.owner.set(self.env().caller());
+            self.max_allowed_percentage.set(max_percentage);
+            self.allow_primary_issuance.set(primary_issuance);
+        }
+
+        /// This function is used to verify transfers initiated by the
+        /// runtime assets
+        ///
+        /// # Arguments
+        /// * `from` - Identity Id of the sender.
+        /// * `to` - Identity Id of the receiver.
+        /// * `value` - Asset amount need to transfer to the receiver.
+        /// * `balance_from` - Balance of sender at the time of transaction.
+        /// * `balance_to` - Balance of receiver at the time of transaction.
+        /// * `total_supply` - Total supply of the asset
+        #[ink(message)]
+        fn verify_transfer(
+            &self,
+            from: Option<IdentityId>,
+            to: Option<IdentityId>,
+            value: Balance,
+            balance_from: Balance,
+            balance_to: Balance,
+            total_supply: Balance,
+        ) -> RestrictionResult {
+            if from == None && *self.allow_primary_issuance.get()
+                || self._is_exempted_or_not(&(to.unwrap()))
+            {
+                return RestrictionResult::Valid;
+            }
+            if ((balance_to + value) * 10u128.pow(6)) / total_supply
+                > *self.max_allowed_percentage.get()
+            {
+                return RestrictionResult::Invalid;
+            } else {
+                return RestrictionResult::Valid;
+            }
```

Comment:
This can be compressed to a single `if ... {return valid} return invalid `

Reply:
I am confused about how it can be compressed into one?

Reply:
```
if from.is_none() && *self.allow_primary_issuance.get()
    || self._is_exempted_or_not(&(to.unwrap_or_default()))
    || ((balance_to + value) * 10u128.pow(6))/total_supply <= *self.max_allowed_percentage.get()
{
    return RestrictionResult::Valid;
}
RestrictionResult::Invalid
```

Reply:
Are you sure for these changes it seems a little bit complex and hard to read?

Reply:
I've just moved your second if statement condition to the first if statement. Once you fix the formatting, it should become more readable :).